File: src/tool/DevilDaggersInfo.Tool.DistributeApp/ProjectReader.fs
module DevilDaggersInfo.Tool.DistributeApp.ProjectReader

open System.IO
open System.Xml

let readVersionFromProjectFile projectFilePath =
    let doc = XmlDocument() in
    doc.LoadXml(File.ReadAllText(projectFilePath))

    let versionList = doc.SelectNodes "/Project/PropertyGroup/Version/text()"
    match versionList.Count with
    | 0 -> failwith "Version not found in project file"
    | _ -> versionList.Item(0).Value

File: src/dev-util/DevilDaggersInfo.DevUtil.DistributeApp/ProjectReader.cs
using System.Xml;

namespace DevilDaggersInfo.DevUtil.DistributeApp;

public static class ProjectReader
{
	public static string ReadVersionFromProjectFile(string projectFilePath)
	{
		XmlDocument doc = new();
		doc.LoadXml(File.ReadAllText(projectFilePath));

		string? version = doc.SelectNodes("/Project/PropertyGroup/Version/text()")?.Item(0)?.InnerText;
		if (version == null)
			throw new("Version not found in project file.");

		return version;
	}
}
